### PR TITLE
Add FlattenedArrayObjectScanner

### PR DESCRIPTION
**Pull request by youngar**
- Use flattened size when iterating flattened arrays
- Use object ptr in Mixed and Reference scanner constructors
- Add more value type array tests
- Add tests for all assorted value alignment types
- Make sure tests are validating after a GC

cc @dmitripivkine

### CODE REVIEW

**Comment by youngar** [comment]
WIP until I run internal testing

**Comment by dmitripivkine** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Comment by dmitripivkine** [comment]
Jenkins test sanity all jdk11

File: runtime/gc_glue_java/FlattenedArrayObjectScanner.hpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	 * @param env The scanning thread environment
+	 * @param allocSpace pointer to space within which the scanner should be instantiated (in-place)
+	 * @param splitAmount The maximum number of array elements to include
+	 * @return Pointer to split scanner in allocSpace
+	 */
+	GC_IndexableObjectScanner *
+	splitTo(MM_EnvironmentBase *env, void *allocSpace, uintptr_t splitAmount)
+	{
+		Assert_MM_unimplemented();
+		return NULL;
```
Comment (youngar) [inline runtime/gc_glue_java/FlattenedArrayObjectScanner.hpp:179]: This one line was added.

**Comment by youngar** [comment]
It failed to compile internally, I think I need to rebase.  I'll launch another internal build based on a rebased branch.

**Comment by youngar** [comment]
Internal builds failed on Friday even after rebasing. I have rebased yet again, and will be launching new builds.

**Comment by youngar** [comment]
Internal builds passed 100%, removing WIP.

**Comment by dmitripivkine** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Comment by dmitripivkine** [comment]
Jenkins test sanity zLinux,aix,win jdk11

**Comment by dmitripivkine** [comment]
Jenkins compile pLinux jdk14

**Pull request merged**